src/GafferScene/SceneAlgo.cpp

```diff
@@ -403,33 +404,25 @@ class CapturingMonitor : public Monitor
 		void processStarted( const Process *process ) override
 		{
 			const Plug *p = process->plug();
-			if( !p->parent<ScenePlug>() || p->getName() != m_scenePlugChildName )
+
+			CapturedProcess::Ptr capturedProcess( new CapturedProcess );
+			if( p->parent() && ( !p->parent<ScenePlug>() || p->getName() != m_scenePlugChildName ) )
 			{
-				// Need to hold a lock while inserting a nullptr sentinel in m_processMap
-				// so child Process's can tell they're not being monitored
-				Mutex::scoped_lock lock( m_mutex );
-				m_processMap.insert( ProcessMap::value_type( process, nullptr ) );
-				return;
+				// This isn't a plug we're interested in.  Set a null plug pointer so we won't report
+				// this result, and create a Scope to disable the monitor so that downstream computations
+				// can go faster without needing to access the monitor
+				capturedProcess->plug = nullptr;
```

Comment:
This is redundant - `PlugPtr` implicitly constructs to `nullptr` anyway.

Reply:
It looks like we're still parenting this CapturedProcess to the parent CapturedProcess (L434 below) so that it gets visited by `historyWalk()`. It seems like we could just not parent it and then we can remove the duplicate "is it a plug we care about?" checks from `historyWalk()`?